### PR TITLE
netifd: improve macvlan with vlan support and remove macvlan hotplug script handeling

### PR DESCRIPTION
**Pull request by sduponch**
- Add support of macvlan directly in netifd
- Remove mac vlan hotplug scripts handler
- Allow direct VLAN allocation on master interface (eth.11 -> macvlan)

know issue : 
- ~A virtual mac address is not generated for new interfaces~
- ~Macvlan hotplug scripts are not deleted wuth a sysupgrade because they are embeded in backup.~
- eth0 take a generated mac address on first boot (but not a duplicated one) => this is because netifd setup the mac address to eth0 before the lan interface get configurer as macvlan (so it does not hurt)

### CODE REVIEW

**Comment by sduponch** [comment]
Now ubus is aware that device is a mac vlan
```
root@sticky:~# ubus call network.device status '{ "name": "lan" }'
{
	"external": false,
	"present": true,
	"type": "MAC VLAN",
	"up": true,
	"carrier": true,
	"parent": "eth0",
	"link-advertising": [
		"10H",
		"10F",
		"100H",
		"100F",
		"1000F"
	],
	"link-supported": [
		"10H",
		"10F",
		"100H",
		"100F",
		"1000F"
	],
	"speed": "1000F",
	"macaddr": "42:42:42:d3:ea:60",
	"statistics": {
		"collisions": 0,
		"rx_frame_errors": 0,
		"tx_compressed": 0,
		"multicast": 0,
		"rx_length_errors": 0,
		"tx_dropped": 0,
		"rx_bytes": 60,
		"rx_missed_errors": 0,
		"tx_errors": 0,
		"rx_compressed": 0,
		"rx_over_errors": 0,
		"tx_fifo_errors": 0,
		"rx_crc_errors": 0,
		"rx_packets": 1,
		"tx_heartbeat_errors": 0,
		"rx_dropped": 0,
		"tx_aborted_errors": 0,
		"tx_packets": 5,
		"rx_errors": 0,
		"tx_bytes": 2154,
		"tx_window_errors": 0,
		"rx_fifo_errors": 0,
		"tx_carrier_errors": 0
	}
}
```
Before my patch : 
```
root@sticky:~# ubus call network.device status '{ "name": "lan" }'
{
	"external": false,
	"present": true,
	"type": "Network device",
	"up": true,
	"carrier": true,
	"link-advertising": [
		"10H",
		"10F",
		"100H",
		"100F",
		"1000F"
	],
	"link-supported": [
		"10H",
		"10F",
		"100H",
		"100F",
		"1000H",
		"1000F"
	],
	"speed": "1000F",
	"macaddr": "42:42:42:7a:f9:c6",
	"statistics": {
		"collisions": 0,
		"rx_frame_errors": 0,
		"tx_compressed": 0,
		"multicast": 7460,
		"rx_length_errors": 0,
		"tx_dropped": 0,
		"rx_bytes": 60459884,
		"rx_missed_errors": 0,
		"tx_errors": 0,
		"rx_compressed": 0,
		"rx_over_errors": 0,
		"tx_fifo_errors": 0,
		"rx_crc_errors": 0,
		"rx_packets": 383151,
		"tx_heartbeat_errors": 0,
		"rx_dropped": 266,
		"tx_aborted_errors": 0,
		"tx_packets": 430575,
		"rx_errors": 0,
		"tx_bytes": 344847898,
		"tx_window_errors": 0,
		"rx_fifo_errors": 0,
		"tx_carrier_errors": 0
	}
}
```

**Comment by sduponch** [comment]
By extension we can now manage a VLAN inside a mac vlan : 
```
config interface 'macvlan1'
        option ifname 'eth0.11'
        option type 'macvlan'
        option proto 'static'
        option macaddr '42:42:42:3f:3f:3f'
        option ipaddr '192.168.0.2'
        option netmask '255.255.255.0'
```
# ifconfig
```
eth0      Link encap:Ethernet  HWaddr 08:00:27:96:D4:90  
          inet6 addr: fe80::a00:27ff:fe96:d490/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:213 errors:0 dropped:0 overruns:0 frame:0
          TX packets:218 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:43303 (42.2 KiB)  TX bytes:28677 (28.0 KiB)

eth0.11   Link encap:Ethernet  HWaddr 08:00:27:96:D4:90  
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0 
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)

macvlan1  Link encap:Ethernet  HWaddr 42:42:42:3F:3F:3F  
          inet addr:192.168.0.2  Bcast:192.168.0.255  Mask:255.255.255.0
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0 
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)
```
ip link show
```
7: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP mode DEFAULT group default qlen 1000
    link/ether 08:00:27:96:d4:90 brd ff:ff:ff:ff:ff:ff
15: eth0.11@eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP mode DEFAULT group default 
    link/ether 08:00:27:96:d4:90 brd ff:ff:ff:ff:ff:ff
16: macvlan1@eth0.11: <BROADCAST,MULTICAST,UP,LOWER_UP,80000> mtu 1500 qdisc noqueue state UNKNOWN mode DEFAULT group default 
    link/ether 42:42:42:3f:3f:3f brd ff:ff:ff:ff:ff:ff
```

**Comment by sduponch** [comment]
ubus call network.device status '{ "name": "eth0.11" }'
```
{
	"external": false,
	"present": true,
	"type": "VLAN",
	"up": true,
	"carrier": true,
	"link-advertising": [
		"10H",
		"10F",
		"100H",
		"100F",
		"1000F"
	],
	"link-supported": [
		"10H",
		"10F",
		"100H",
		"100F",
		"1000F"
	],
	"speed": "1000F",
	"statistics": {
		"collisions": 0,
		"rx_frame_errors": 0,
		"tx_compressed": 0,
		"multicast": 0,
		"rx_length_errors": 0,
		"tx_dropped": 0,
		"rx_bytes": 0,
		"rx_missed_errors": 0,
		"tx_errors": 0,
		"rx_compressed": 0,
		"rx_over_errors": 0,
		"tx_fifo_errors": 0,
		"rx_crc_errors": 0,
		"rx_packets": 0,
		"tx_heartbeat_errors": 0,
		"rx_dropped": 0,
		"tx_aborted_errors": 0,
		"tx_packets": 0,
		"rx_errors": 0,
		"tx_bytes": 0,
		"tx_window_errors": 0,
		"rx_fifo_errors": 0,
		"tx_carrier_errors": 0
	}
}
```

File: overthebox/defaults/network.defaults [hunk 1]
```diff
@@ -10,6 +10,17 @@ if [ $? -ne 0 ]; then
 EOF
 fi
 
+MACPREFIX="42:42:42"
+if ! uci -q get network.lan.macaddr 2>/dev/null; then
+	MACSUFFIX=$(hexdump -n3 -e'3/1 ":%02x"' /dev/urandom)
```
Comment (PouuleT) [inline overthebox/defaults/network.defaults]: We already had issues with reads of /dev/urandom
We should always double check if it's ok or not

**Pull request closed without merging**